core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -128,17 +130,16 @@ private DeciderOutcome decide(final WorkflowDef workflowDef, final Workflow work
                 .map(Task::getReferenceTaskName)
                 .collect(Collectors.toSet());
 
-        Map<String, Task> tasksToBeScheduled = new LinkedHashMap<>();
-
-        preScheduledTasks.forEach(pst -> {
-            tasksToBeScheduled.put(pst.getReferenceTaskName(), pst);
-        });
+        //Traverse the pre-scheduled tasks to a linkedHasMap
+        Map<String, Task> tasksToBeScheduled = preScheduledTasks.stream()
+                .collect(Collectors.toMap(Task::getReferenceTaskName, Function.identity(),
+                        (element1, element2) -> element2, LinkedHashMap::new));
```

Comment:
should throwingMerger be used here?

Reply:
That should be the case, but the existing code was just overwriting the value, kept the behavior of the existing code.